lib/utilities/paginator.rb

```diff
@@ -0,0 +1,106 @@
+module Utilities
+  module Paginator
+    def items_to_display(model:, items_per_page: 5, current_page: 1)
+      @items_per_page = items_per_page
+      @current_page = current_page
+      @number_of_items = model.count
+      @last_page = (@number_of_items.to_f / @items_per_page).ceil
+      model.offset(items_per_page * (@current_page - 1))
+           .limit(items_per_page)
+    end
+
+    def paginator(max_page_blocks: 7)
+      render partial: 'shared/paginator',
+             locals: { blocks: pages_to_display(@current_page,
+                                                max_page_blocks,
+                                                @last_page),
+                       current_page: @current_page,
+                       previous_link_css: previous_link_css,
+                       next_link_css: next_link_css,
+                       last_page: @last_page }
+    end
+
+    def page_item_css(page, current_page)
+      if page == current_page
+        'page-item active'
+      elsif page.to_s == '...'
```

Comment:
You use the ellipses (```...```) in several places - it might be worth making it a constant, in case you ever want to change it in the future. This will also allow you to give it a more descriptive/semantic name so that you don't have 'Magic Numbers' (or text in this case) in your code.